app/models/miq_schedule.rb

```diff
@@ -192,7 +192,11 @@ def action_vm_scan(obj, _at)
 
   def action_scan(obj, _at)
     sched_action[:options] ||= {}
-    obj.scan
+    if obj.respond_to?(:scan_with_userid)
```

Comment:
does it make sense to encapsulate this `If branch` to the `scan` method ( and thanks to that have here only calling `scan` method )

Reply:
What other objects is this called on?  Can we just always pass the userid?

Reply:
@agrare `EmsCluster` also calls `scan` without `userid`, and without this `if` I would break their scan (to be honest I'm not acquainted with this entity at all.)

Reply:
> Can/should we add an optional ignored userid arg to it?

Yeah I think so, all of the other scan methods are `def scan(userid = "system", options = {})` so this would keep things consistent.

> Anybody knows what that is (and what's save_drift_state)?

http://manageiq.org/docs/reference/fine/doc-Managing_Infrastructure_and_Inventory/miq/#detecting-drift-on-clusters
I'm not sure if this can be scheduled but it can be run from the UI.

> "the best way to get the right answer on the internet is not to ask a question; it's to post the wrong answer."

LOL that is brilliant

Reply:
And EmsCluster#scan